proposals/0000-improved-ghc-development.rst

```diff
@@ -0,0 +1,216 @@
+
+  With the ability to login directly on Trac using your GitHub account, I believe it's easier to take issues on one place: Trac. So read-only would be enough.
+
+  As a whole, I find Trac MUCH better for ticket triaging than Github or Phab,
+  both of which seem to be quite bare and simple in what they provide. I am not
+  in favor of ditching it. Whatever we gain we lose in the ability to generate custom views on issues. Which is a powerful filtering mechanism for prioritization.
+
+  Also we have and continue to accept patches just uploaded
+  to Trac as a diff. We tend to ask people to upload it to phab for better reviews
+  and so it's attributed to them when we commit. Some don't (and we then do it ourselves),
+  most due. If they don't need another login then I suspect almost all would.
+
+  There's a (seemingly) actively maintained project that does all the above, could we leverage it?
+  https://github.com/trac-hacks/trac-github
+
+- Allow doc changes on Trac
+
+  I am not sure about the feasibility of this one, but there is a Trac plugin that adds a new section to trac under /doc that is able to render .rst files and more from a source repository. Essentially making a wiki from them.
+
+  https://github.com/trac-hacks/tracdocs
+
+  We could have harbormaster build the docs and provide them to trac. I am not sure how reviews would go. Styling would be another issue. 
+
+  Lastly this plugin seems to only support SVN. but that's not a huge barrier.
+
+- Helping Newcomers
+
+   * Create a better landing page for newcomers (see proposal below)
+   * Show newcomers tickets more prominently
+   * Clean up build instructions. The pages have become way too complex over the years. 
+   * Create native setup script particular to an environment.
+
+  As I am a windows maintainer, I will offer a proposal: Use scripts similar to my own setup scripts. my setup script for a 100% unattended build env setup for Windows are here: https://github.com/Mistuke/GhcDevelChoco/releases
+
+  These are entirely self contained environments that can be removed by a simple rm -rf /.
+  You can have as many as you want on the same machine without them interfering with eachother or with whatever else you might have done to your GHC already installed.
+   
+  It's not 100% production ready but it works and does so well.
+
+  This script will setup an environment containing (if you want):
+   * A full msys setup
+   * gdb/gcc etc for debugging
+   * check out Haskell sources
+   * do a local install of php and arc
+   * configure arc
+   * setup Hadrian
+   * setup SSH (for those who want to interface with it this way)
+ 
+  The script is set up to provide you with a working environment with little interaction needed. Setup is just one call. 
+
+  For Windows, such a script is a much better solution than Docker, simply because Docker does not work very well for Windows targets.
```

Comment:
Good to know; this is likely what I will deploy on our Windows Harbormaster target.